zhaquirks/tuya/builder/__init__.py

```diff
@@ -0,0 +1,493 @@
+        endpoint_id: Optional[int] = None,
+        dp_handler: str = "_dp_2_attr_update",
+        type: type = t.uint16_t,
+        access: foundation.ZCLAttributeAccess = foundation.ZCLAttributeAccess.NONE,
+        is_manufacturer_specific=True,
+    ) -> QuirkBuilder:  # fmt: skip
+        """Add an Tuya DataPoint and corresponding AttributeDef."""
+        self.tuya_attribute(
+            dp_id=dp_id,
+            attribute_name=attribute_name,
+            type=type,
+            access=access,
+            is_manufacturer_specific=is_manufacturer_specific,
+        )
+        self.tuya_dp(
+            dp_id=dp_id,
+            ep_attribute=ep_attribute,
+            attribute_name=attribute_name,
+            dp_converter=dp_converter,
+            converter=converter,
+            endpoint_id=endpoint_id,
+            dp_handler=dp_handler,
+        )
+        return self
+
+    def tuya_switch(
+        self,
+        dp_id: int,
+        attribute_name: str = "on_off",
+        endpoint_id: int = 1,
+        force_inverted: bool = False,
+        invert_attribute_name: str | None = None,
+        off_value: int = 0,
+        on_value: int = 1,
+        entity_platform=EntityPlatform.SWITCH,
+        entity_type: EntityType = EntityType.CONFIG,
+        initially_disabled: bool = False,
+        attribute_initialized_from_cache: bool = True,
+        translation_key: str | None = None,
+        fallback_name: str | None = None,
+    ) -> QuirkBuilder:
+        """Add an EntityMetadata containing SwitchMetadata and return self.
+
+        This method allows exposing a switch entity in Home Assistant.
+        """
+        self.tuya_dp_attribute(
+            dp_id=dp_id,
+            attribute_name=attribute_name,
+            type=t.Bool,
+            access=foundation.ZCLAttributeAccess.Read
+            | foundation.ZCLAttributeAccess.Write,
+        )
+        self.switch(
+            attribute_name=attribute_name,
+            cluster_id=TUYA_CLUSTER_ID,
+            endpoint_id=endpoint_id,
+            force_inverted=force_inverted,
+            invert_attribute_name=invert_attribute_name,
+            off_value=off_value,
+            on_value=on_value,
+            entity_platform=entity_platform,
+            entity_type=entity_type,
+            initially_disabled=initially_disabled,
+            attribute_initialized_from_cache=attribute_initialized_from_cache,
+            translation_key=translation_key,
+            fallback_name=fallback_name,
+        )
+        return self
+
+    def tuya_enum(
+        self,
+        dp_id: int,
+        attribute_name: str,
+        enum_class: type[Enum],
+        access: foundation.ZCLAttributeAccess = foundation.ZCLAttributeAccess.Read
+        | foundation.ZCLAttributeAccess.Write,
+        endpoint_id: int = 1,
+        entity_type: EntityType = EntityType.CONFIG,
+        entity_platform: EntityPlatform = EntityPlatform.SELECT,
```

Comment:
Doesn't really matter, but let's swap these around (like in https://github.com/zigpy/zigpy/pull/1487), so `entity_platform` is always before `entity_type`.
```suggestion
        entity_platform: EntityPlatform = EntityPlatform.SELECT,
        entity_type: EntityType = EntityType.CONFIG,
```